core/src/main/scala/kafka/controller/PartitionStateMachine.scala

```diff
@@ -435,9 +435,10 @@ class PartitionStateMachine(config: KafkaConfig,
 }
 
 object PartitionLeaderElectionAlgorithms {
-  def offlinePartitionLeaderElection(assignment: Seq[Int], isr: Seq[Int], liveReplicas: Set[Int], uncleanLeaderElectionEnabled: Boolean): Option[Int] = {
+  def offlinePartitionLeaderElection(assignment: Seq[Int], isr: Seq[Int], liveReplicas: Set[Int], uncleanLeaderElectionEnabled: Boolean, controllerContext: ControllerContext): Option[Int] = {
     assignment.find(id => liveReplicas.contains(id) && isr.contains(id)).orElse {
       if (uncleanLeaderElectionEnabled) {
+        controllerContext.stats.uncleanLeaderElectionRate.mark()
```

Comment:
We probably only want to mark the metric if assignment.find(liveReplicas.contains) is not empty.

Reply:
@junrao Thanks for the review. Updated the PR.